stablehlo/dialect/Base.cpp

```diff
@@ -612,10 +612,26 @@ SmallVector<int64_t> getI64Array(Attribute attr) {
   if (auto array = attr.dyn_cast<DenseI64ArrayAttr>())
     return llvm::to_vector(array.asArrayRef());
   llvm::report_fatal_error(
-      "called i64ArrayOrElementsValues on Attribute that was neither a "
+      "called getI64Array on Attribute that was neither a "
       "DenseIntElementsAttr or a DenseI64ArrayAttr",
       false);
 }
 
+SmallVector<bool> getBoolArray(Attribute attr) {
+  if (!attr) return {};
+  if (auto elements = attr.dyn_cast<DenseIntOrFPElementsAttr>())
+    return llvm::to_vector(elements.getValues<bool>());
+  if (auto array = attr.dyn_cast<DenseBoolArrayAttr>()) {
+    // TODO: find a cleaner way
+    SmallVector<bool> vec;
```

Comment:
`return array.asArrayRef().vec()` ?

Reply:
Sorry, meant to fix this before re-assigning you. Thanks for the rec! That returns a `std::vector` so it can't be used as is. The root problem was that I was trying to do `llvm::to_vector(array.asArrayRef())` and that fails because of the way that's implemented:

```
llvm-project/llvm/include/llvm/ADT/SmallVector.h +1308:11: error: type 'decltype(__cont.begin())' (aka 'const bool *') cannot be narrowed to 'bool' in initializer list [-Wc++11-narrowing]
  return {std::begin(Range), std::end(Range)};
          ^~~~~~~~~~~~~~~~~
stablehlo/dialect/Base.cpp +625:18: note: in instantiation of function template specialization 'llvm::to_vector<llvm::ArrayRef<bool>>' requested here
    return llvm::to_vector(array.asArrayRef());
                 ^
llvm-project/llvm/include/llvm/ADT/SmallVector.h +1308:11: note: insert an explicit cast to silence this issue
  return {std::begin(Range), std::end(Range)};
          ^~~~~~~~~~~~~~~~~
          static_cast<bool>( )
```

(For some reason the compiler thinks the code is trying to use the `initializer_list` constructor of `SmallVector` and it "helpfully" suggests casting the iterators to `bool`.)

I decided to just be explicit: `return SmallVector<bool>(array.asArrayRef());`

Also while looking into this I realized the new attr was not tested so I added two basic tests (a convolution with dense window reversal and another with array window reversal).